File: cpp/cw_funkcje02.cpp
#include <iostream>
#include <algorithm>
using namespace std;


double srednia(int suma, int n){
    suma = suma / n;
    return suma;

}

double mediana(int* liczba, int n){
    sort(liczba, liczba + n);
    if (n % 2 == 0) {
        return (liczba[n / 2 - 1] + liczba[n / 2]) / 2;
    } 
    else {
        return liczba[(n) / 2 ];

    }

}

void oceny(int* liczba, int n){
    
    cout << "Ocena 1 występuje " << count( liczba, liczba + n, 1 ) << " razy" << endl;
    cout << "Ocena 2 występuje " << count( liczba, liczba + n, 2 ) << " razy" << endl;
    cout << "Ocena 3 występuje " << count( liczba, liczba + n, 3 ) << " razy" << endl;
    cout << "Ocena 4 występuje " << count( liczba, liczba + n, 4 ) << " razy" << endl;
    cout << "Ocena 5 występuje " << count( liczba, liczba + n, 5 ) << " razy" << endl;
    cout << "Ocena 6 występuje " << count( liczba, liczba + n, 6 ) << " razy" << endl;
}

int main(int argc, char **argv){

    int n;
    double suma;
    cout << "Podaj ilość ocen: "<< endl;
    cin >> n;
    if( n > 0 && n < 20){
        int * liczba = new int[n];
        cout << "Podaj oceny (Prawidłowe oceny to 1,2,3,4,5,6 ):" << endl;
        for (int i = 0; i < n; i++){
            cin >> liczba[i];
            if (liczba[i] < 1 || liczba[i] > 6){
                cout << "Podałeś złą wartość!";
                return 0;
            }  
            suma += liczba[i];
        }
        cout << "Mediana wynosi: " << mediana(liczba, n) << endl;
        cout << "Średnia wynosi: " << srednia(suma, n) << endl;
        oceny(liczba, n);
    }
    else {
        cout << "błędna wartość!";
    }



}

File: pajton/cw_funkcje02.py


def srednia(a):
   suma = sum(a)
   return suma / float(len(a))


def mediana(b):

    b = sorted(b)
    if  len(b) % 2 == 0:
        pol = int(len(b) / 2)
        return float(sum(b[pol - 1:pol + 1])) / 2.0
    else:
        return b[int(len(b) / 2)]

def oceny(c):
    for i in c:
        licznik = c.count(i)
        print("Ocena", i, "występuje", licznik, "razy")
    return 0


def main(args):

    n = int(input("Podaj ilość ocen: "))
    m = n
    if n <20 and n>0:
      lista = []
      i = 1
      while i < m + 1:
        liczby = int(input("Podaj oceny: "))
        if liczby > 0 and liczby < 7:
          lista.append(liczby)
          i += 1
        else:
          print("Błędna wartość!")
      print("Średnia wynosi", srednia(lista))
      print("Mediana wynosi:",mediana(lista)) 
      oceny(lista)

    else:
      print("Błędna wartość!")
    return 0

if __name__ == '__main__':
    import sys
    sys.exit(main(sys.argv))
